massa-models/src/operation.rs

```diff
@@ -972,13 +970,13 @@ impl SecureShareOperation {
     }
 
     /// Get the max amount of gas used by the operation (`max_gas`)
-    pub fn get_gas_usage(&self) -> u64 {
+    pub fn get_gas_usage(&self, gas_usage: u64) -> u64 {
         match &self.content.op {
             OperationType::ExecuteSC { max_gas, .. } => *max_gas,
             OperationType::CallSC { max_gas, .. } => *max_gas,
-            OperationType::RollBuy { .. } => 0,
-            OperationType::RollSell { .. } => 0,
-            OperationType::Transaction { .. } => 0,
+            OperationType::RollBuy { .. } => gas_usage,
+            OperationType::RollSell { .. } => gas_usage,
+            OperationType::Transaction { .. } => gas_usage,
```

Comment:
Maybe the designed changed in the meantime, but I though that the gas_usage would also be added for the ExecuteSC and CallCS, no?

Because there are still some things we do when including SC operations that would not be taken into accout in the (user declared) max_gas.

Reply:
You're right!!! I've addressed, so far, only the basic operations.
Partly because:
1. I was (am) not that satisfied by the design (how the `gas_usage` in injected from it's constant definition to its point of use). But doing better would involve too big refactor.
2. I have to check if adding `gas_usage` does not conflict with what Thomas is doing #4483

Reply:
Ok it makes sence, thanks!